jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -204,8 +230,105 @@ public static PluginConfigurationProcessor processCommonConfiguration(
       jibContainerBuilder.setCreationTime(Instant.now());
     }
 
+    PluginConfigurationProcessor.configureContainerizer(
+        containerizer, rawConfiguration, projectProperties);
+
     return new PluginConfigurationProcessor(
-        jibContainerBuilder, baseImageReference, optionalFromCredential.isPresent());
+        jibContainerBuilder,
+        containerizer,
+        baseImageReference,
+        targetImageReference,
+        isBaseImageCredentialPresent,
+        isTargetImageCredentialPresent);
+  }
+
+  /**
+   * Gets the value of the {@code appRoot} parameter. If the parameter is empty, returns {@link
+   * JavaLayerConfigurations#DEFAULT_WEB_APP_ROOT} for WAR projects or {@link
+   * JavaLayerConfigurations#DEFAULT_APP_ROOT} for other projects.
+   *
+   * @param rawConfiguration raw configuration data
+   * @param projectProperties used for providing additional information
+   * @return the app root value
+   * @throws AppRootInvalidException if {@code appRoot} value is not an absolute Unix path
+   */
+  @VisibleForTesting
+  static AbsoluteUnixPath getAppRootChecked(
+      RawConfiguration rawConfiguration, ProjectProperties projectProperties)
+      throws AppRootInvalidException {
+    String appRoot = rawConfiguration.getAppRoot();
+    if (appRoot.isEmpty()) {
+      appRoot =
+          projectProperties.isWarProject()
+              ? JavaLayerConfigurations.DEFAULT_WEB_APP_ROOT
+              : JavaLayerConfigurations.DEFAULT_APP_ROOT;
+    }
+    try {
+      return AbsoluteUnixPath.get(appRoot);
+    } catch (IllegalArgumentException ex) {
+      throw new AppRootInvalidException(appRoot, appRoot, ex);
+    }
+  }
+
+  @FunctionalInterface
+  private static interface InferredAuthProvider {
```

Comment:
`static` unnecessary

Reply:
interfaces and classes should go above methods at top of class

Reply:
Oh, I think I have not been following the repo style. I've been putting priority `public`/`private`.. over anything (even if they are classes).

Reply:
I think interfaces are just implicitly static?